data_collection.py

```diff
@@ -13,8 +13,10 @@ def store_query(input_code, output_code, input_lang, output_lang):
         print("Please set GOOGLE_APPLICATION_CREDENTIALS environmental variable. See readme for more info")
         return
     
-    cred = credentials.Certificate(cred_object)
-    firebase_admin.initialize_app(cred)
+    # only initialize firebase once
+    if not firebase_admin._apps:
```

Comment:
I would wrap this function in a `DataService` [singleton](https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html) class. That way it is only initialized once. Happy to hop on a call and walk through this!